### DISCUSSION TITLE
フロントエンドの例外処理の実装方針について知りたい

### DISCUSSION THREAD

**Discussion by KentaHizume** [AlesInfiny/maris#1451] [05: Q&A] [answered]
# 質問
DresscaフロントエンドでHTTPステータスコードに応じた例外処理をしたい場合、どの場所に実装したらよいか教えていただけないでしょうか。

# 質問のモチベーション
#1237 において、Dresscaに、楽観同時実行制御の機能の実装を考えているためです。
具体的には、APIコールの結果、[409 Conflict](https://developer.mozilla.org/ja/docs/Web/HTTP/Status/409) が返ってきた場合に、ユーザに通知を行い、画面のリロードを行いたいです。

**Comment by tsuna-can-se** [comment]
#795 にて実装方針を検討中です。

**Comment by 1nu1taichi** [comment, marked as answer]
リロードをする、という条件においては axios.inceptors.response 内でも実装できます。
ただしリロードはすべての store もリフレッシュされてしまうため、注意が必要です。リフレッシュが目的であれば、個別で特定の store だけを fetch し直す方が健全かと思いますがいかがでしょうか。
現在検討中の方法では、 axios.interceptors.response 内でステータスコードに応じたカスタムエラーに詰め、エラーハンドラーで共通処理を実現する、という方法を取っています。
